__init__.py

```diff
@@ -2678,699 +3035,720 @@ def get_eigenvalues(self, kpt=None, spin=None, efermi=None):
 
         eig = []
         for k in kp:
-            f = open(self.scratch+'/calc.save/'+k, 'r')
-            a = f.readline()
-            while a.upper().find('<EIG')<0:
-                a = f.readline()
+            f = open(self.scratch + '/calc.save/' + k, 'r')
+            a = f.readline().decode('utf-8')
+            while a.upper().find('<EIG') < 0:
+                a = f.readline().decode('utf-8')
             nbnd = int(a.split('"')[-2])
-            eig.append(hartree*np.fromfile(f, dtype=float, count=nbnd, sep=' ') - ef)
+            eig.append(Hartree * np.fromfile(
+                f, dtype=float, count=nbnd, sep=' ') - ef)
             f.close()
 
-        spinall = spin not in ('up','down',0,1)
+        spinall = spin not in ('up', 'down', 0, 1)
         if kpt is not None and spinall:
             return np.array(eig[0])
         elif kpt is None and spinall and self.spinpol:
-            return np.reshape(np.array(eig), (2,nkp2,nbnd))
+            return np.reshape(np.array(eig), (2, nkp2, nbnd))
         else:
             return np.array(eig)
 
-
-
     def read_3d_grid(self, stream, log):
-        f = open(self.localtmp+'/'+log, 'a')
-        x = stream.readline()
-        while x!='' and x[:11]!='DATAGRID_3D':
-            f.write(x)
-            x = stream.readline()
-        if x=='':
-            raise RuntimeError, 'error reading 3D data grid'
-        f.write(x)
-        nx, ny, nz = [int(y) for y in stream.readline().split()]
-        origin = np.array([float(y) for y in stream.readline().split()])
-        cell = np.empty((3,3), np.float)
-        for i in range(3):
-            cell[i][:] = [float(y) for y in stream.readline().split()]
-        data = np.reshape(np.fromfile(stream, count=nx*ny*nz, sep=' '),
-            (nx,ny,nz), order='F')
-
-        x = stream.readline()
-        while x!='':
+        with open(self.localtmp + '/' + log, 'a') as f:
+            x = stream.readline().decode('utf-8')
+            while x != '' and x[:11] != 'DATAGRID_3D':
+                f.write(x)
+                x = stream.readline().decode('utf-8')
+            if x == '':
+                raise RuntimeError('error reading 3D data grid')
             f.write(x)
-            x = stream.readline()
-
-        f.close()
-        return (origin,cell,data)
 
+            n = [int(_) for _ in stream.readline().split()]
+            origin = np.array([float(_) for _ in stream.readline().split()])
 
-    def read_2d_grid(self, stream, log):
```

Comment:
This function isn't called at all. I removed it, but can add it back if necessary.